images/shadowsocks-rust/main.tf

```diff
@@ -0,0 +1,51 @@
+terraform {
+  required_providers {
+    oci = { source = "chainguard-dev/oci" }
+  }
+}
+
+variable "target_repository" {
+  description = "The docker repo into which the image and attestations should be published."
+}
+
+locals {
+  components = toset(["sslocal", "ssserver"])
+}
+
+module "config" {
+  for_each = local.components
+  source   = "./configs"
+  name     = each.key
+}
+
+module "latest" {
+  for_each = local.components
+  source   = "../../tflib/publisher"
+
+  name              = basename(path.module)
+  target_repository = var.target_repository
```

Comment:
we should add each component as suffix to the repository:

`"${var.target_repository}-${each.key}"`